sdk/core/azure-core/inc/azure/core/nullable.hpp

```diff
@@ -261,10 +261,72 @@ namespace Azure { namespace Core {
       return std::move(m_value);
     }
 
+    // observers
+
     /**
      * @brief `operator bool` on the condition of #Azure::Core::Nullable::HasValue.
      */
-    explicit operator bool() const noexcept { return HasValue(); }
+    constexpr explicit operator bool() const noexcept { return HasValue(); }
+
+    /**
+     * @brief Accesses the contained value.
+     * @return Returns a pointer to the contained value
+     * @warning The behavior is undefined if `*this` does not contain a value.
+     * @note This operator does not check whether the optional contains a value!
+             You can do so manually by using #HasValue() or simply operator #bool().
+             Alternatively, if checked access is needed, #GetValue() or #ValueOr() may be used.
```

Comment:
What's `ValueOr()`? Are we planning to add `ValueOrDefault`. Is that what this is supposed to refer to?

Reply:
ValueOr() is the method that exists to return the value or whatever you specify

Reply:
Ah yes, missed that.